lib/util/git/url.ts

```diff
@@ -1,13 +1,37 @@
-import GitUrlParse from 'git-url-parse';
+import gitUrlParse from 'git-url-parse';
 import { logger } from '../../logger';
+import { detectPlatform } from '../common';
 import * as hostRules from '../host-rules';
 import { regEx } from '../regex';
 
+export function parseGitUrl(url: string): gitUrlParse.GitUrl {
+  const parsed = gitUrlParse(url);
+
+  // Workaround for https://github.com/IonicaBizau/parse-path/issues/38
+  if (parsed.port && parsed.resource.endsWith(`:${parsed.port}`)) {
+    parsed.resource = parsed.resource.substring(
+      0,
+      parsed.resource.length - `:${parsed.port}`.length
+    );
+  }
+
+  return parsed;
+}
+
 export function getHttpUrl(url: string, token?: string): string {
-  const parsedUrl = GitUrlParse(url);
+  const parsedUrl = parseGitUrl(url);
 
   parsedUrl.token = token ?? '';
 
+  if (token) {
+    switch (detectPlatform(url)) {
+      case 'gitlab':
+        parsedUrl.token = token.includes(':')
+          ? token
+          : `gitlab-ci-token:${token}`;
```

Comment:
Are you sure the username is not `oauth2` instead of `gitlab-ci-token`? That's the username I used in [my workaround](https://github.com/renovatebot/renovate/issues/16561#issuecomment-1206854917) and that @rndmh3ro confirmed.

Reply:
username doesn't matter, we use the same in other git auth file